netbox_onboarding/netdev_keeper.py

```diff
@@ -0,0 +1,290 @@
+        except AddrFormatError:
+            try:
+                # Do a lookup of name to get the IP address to connect to
+                checked_ip = socket.gethostbyname(self.hostname)
+                self.hostname = checked_ip
+                return True
+            except socket.gaierror:
+                # DNS Lookup has failed, Raise an exception for unable to complete DNS lookup
+                raise OnboardException(
+                    reason="fail-dns",
+                    message=f"ERROR failed to complete DNS lookup: {self.hostname}"
+                )
+
+    def check_reachability(self):
+        """Ensure that the device at the mgmt-ipaddr provided is reachable.
+
+        We do this check before attempting other "show" commands so that we know we've got a
+        device that can be reached.
+
+        Raises:
+          OnboardException('fail-connect'):
+            When device unreachable
+        """
+
+        logging.info("CHECK: IP %s:%s", self.hostname, self.port)
+
+        try:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.settimeout(self.timeout)
+            sock.connect((self.hostname, self.port))
+
+        except (socket.error, socket.timeout, ConnectionError):
+            raise OnboardException(reason="fail-connect",
+                                   message=f"ERROR device unreachable: {self.hostname}:{self.port}")
+
+    def guess_netmiko_device_type(self):
+        """Guess the device type of host, based on Netmiko."""
+        guessed_device_type = None
+
+        remote_device = {
+            "device_type": "autodetect",
+            "host": self.hostname,
+            "username": self.username,
+            "password": self.password,
+            "secret": self.secret,
+        }
+
+        try:
+            logging.info(f"INFO guessing device type: {self.hostname}")
+            guesser = SSHDetect(**remote_device)
+            guessed_device_type = guesser.autodetect()
+            logging.info(f"INFO guessed device type: {guessed_device_type}")
+
+        except NetMikoAuthenticationException as err:
+            logging.error("ERROR %s", err)
+            raise OnboardException(reason="fail-login",
+                                   message=f"ERROR: {str(err)}"
+                                   )
+
+        except (NetMikoTimeoutException, SSHException) as err:
+            logging.error(f"ERROR: {str(err)}")
+            raise OnboardException(reason="fail-connect",
+                                   message=f"ERROR: {str(err)}"
+                                   )
+
+        except Exception as err:
+            logging.error(f"ERROR: {str(err)}", err)
+            raise OnboardException(reason="fail-general",
+                                   message=f"ERROR: {str(err)}")
+
+        logging.info(f"INFO device type is: {guessed_device_type}")
+
+        return guessed_device_type
```

Comment:
It looks like you've lost support for the `platform_map` plugin setting added in #55. Is that an intended behavior change?

Reply:
Thanks! I changed it as follows
```
self.netdev_nb_platform_slug = self.netdev_nb_platform_slug or self.netdev_netmiko_device_type
```

into
```
            self.netdev_nb_platform_slug = self.netdev_nb_platform_slug or \
                                           PLUGIN_SETTINGS["platform_map"].get(self.netdev_netmiko_device_type) or \
                                           self.netdev_netmiko_device_type
```